src/sentry/static/sentry/app/components/activity/item.jsx

```diff
@@ -35,19 +35,18 @@ class ActivityItem extends React.Component {
     this.state = {
       clipped: this.props.defaultClipped,
     };
+    this.activityBubbleRef = React.createRef();
   }
 
   componentDidMount() {
-    if (this.refs.activityBubble) {
-      let bubbleHeight = this.refs.activityBubble.offsetHeight;
+    if (this.activityBubbleRef.current) {
+      let bubbleHeight = this.activityBubbleRef.current.offsetHeight;
 
       if (bubbleHeight > this.props.clipHeight) {
-        /*eslint react/no-did-mount-set-state:0*/
         // okay if this causes re-render; cannot determine until
         // rendered first anyways
-        this.setState({
-          clipped: true,
-        });
+        // eslint-disable-next-line react/no-did-mount-set-state
```

Comment:
Hmm we should get rid of this rule.

Reply:
I think it's a decent warning. You probably don't want to be doing this since it will cause a re-render, but like the comment above says, sometimes you need to re-render.